MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm

```diff
@@ -39,15 +39,15 @@
                                                MVKDescriptorSet* descSet,
                                                MVKShaderResourceBinding& dslMTLRezIdxOffsets,
                                                MVKArrayRef<uint32_t> dynamicOffsets,
-                                               uint32_t* pDynamicOffsetIndex) {
+                                               uint32_t baseDynamicOffsetIndex) {
     if (_isPushDescriptorLayout) return;
 
 	clearConfigurationResult();
     size_t bindCnt = _bindings.size();
     for (uint32_t descIdx = 0, bindIdx = 0; bindIdx < bindCnt; bindIdx++) {
 		descIdx += _bindings[bindIdx].bind(cmdEncoder, descSet, descIdx,
 										   dslMTLRezIdxOffsets, dynamicOffsets,
-										   pDynamicOffsetIndex);
+										   baseDynamicOffsetIndex);
```

Comment:
I'm wondering if we can simplify this overall solution to a single change to this `MVKDescriptorSetLayout::bindDescriptorSet()` function. Something like...

Creating a temp copy of the portion of `dynamicOffsets` that is covered by this descriptor set here, and that is sorted by descriptor order instead of binding number, and then submitting that re-sorted array to the individual bindings here.

Each `MVKBufferDescriptor` would then just do what it currently does, using the next value, and incrementing the overall `pDynamicOffsetIndex` as it does now.

Making the `_bindingToIndex` an ordered `map` instead of `unordered_map` might be a help in the task of re-sorting of the elements of `dynamicOffsets` on the fly (ie- iterate through `_bindingToIndex` and for each dynamic buffer descriptor found, move the next value in `dynamicOffsets` into the temp array).

Thoughts?

Reply:
Thanks for the quick response. Totally your call here. I designed it this way to avoid any memory allocation or sorting during the bind* operation since that is on the critical fast path. I was worried about cases with large amounts of descriptors, and since this solution is just an extra uint32_t stored per descriptor/descriptorArray it seems reasonable.

Reply:
Okay. Yeah...that does make sense.